### DISCUSSION TITLE
Error with Zigbee2MQTT LXC Script

### DISCUSSION THREAD

**Discussion by mablungfr** [tteck/Proxmox#2333] [Help Needed] [answered]
Hello, I'm using Proxmox 8.1.3 and I'm unable to use the Zigbee2MQTT LXC script.
I encounter an error during the execution of the script when configuring zigbee2mqtt, with an error at line 23.
I hace tried default and specific options, same error.
Could someone please try it?

**Comment by tteck** [comment]
The script runs without any problems from this end. 

```
 _____   _       __             ___   __  _______  ____________
/__  /  (_)___ _/ /_  ___  ___ |__ \ /  |/  / __ \/_  __/_  __/
  / /  / / __  / __ \/ _ \/ _ \__/ // /|_/ / / / / / /   / /   
 / /__/ / /_/ / /_/ /  __/  __/ __// /  / / /_/ / / /   / /    
/____/_/\__, /_.___/\___/\___/____/_/  /_/\___\_\/_/   /_/     
       /____/ 🐝
 
Using Advanced Settings
Using Distribution: debian
Using debian Version: 12
Using Container Type: 0
Using Root Password: Automatic Login
Container ID: 111
Using Hostname: zigbee2mqtt
Using Disk Size: 4
Allocated Cores: 2
Allocated RAM: 1024
Using Bridge: vmbr0
Using IP Address: dhcp
Using Gateway IP Address: Default
Using APT-Cacher IP Address: 192.168.86.110
Disable IPv6: yes
Using Interface MTU Size: Default
Using DNS Search Domain: Host
Using DNS Server IP Address: Host
Using MAC Address: AE:1A:60:07:B7:59
Using Vlan: Default
Enable Root SSH Access: no
Enable Verbose Mode: no
Creating a Zigbee2MQTT LXC using the above advanced settings
 ✓ Using local for Template Storage.
 ✓ Using local-lvm for Container Storage.
 ✓ Updated LXC Template List
 ✓ LXC Container 111 was successfully created.
 ✓ Started LXC Container
 ✓ Set up Container OS
 ✓ Network Connected: 192.168.86.253 
 ✓ Internet Connected
 ✓ DNS Resolved github.com to 140.82.114.3
 ✓ Updated Container OS
 ✓ Installed Dependencies
 ✓ Set up Node.js Repository
 ✓ Installed Node.js
 ✓ Set up Zigbee2MQTT Repository
Switch to Edge/dev branch? (y/N) y
 ✓ Installed Zigbee2MQTT
 ✓ Created Service
 ✓ Customized Container
 ✓ Cleaned
 ✓ Completed Successfully!
```

**Reply by mablungfr** [reply]
Hum.. damned. My output:
`_____   _       __             ___   __  _______  ____________
/__  /  (_)___ _/ /_  ___  ___ |__ \ /  |/  / __ \/_  __/_  __/
  / /  / / __  / __ \/ _ \/ _ \__/ // /|_/ / / / / / /   / /   
 / /__/ / /_/ / /_/ /  __/  __/ __// /  / / /_/ / / /   / /    
/____/_/\__, /_.___/\___/\___/____/_/  /_/\___\_\/_/   /_/     
       /____/ 🐝
 
Using Default Settings
Using Distribution: debian
Using debian Version: 12
Using Container Type: 0
Using Root Password: Automatic Login
Using Container ID: 105
Using Hostname: zigbee2mqtt
Using Disk Size: 4GB
Allocated Cores 2
Allocated Ram 1024
Using Bridge: vmbr0
Using Static IP Address: dhcp
Using Gateway IP Address: Default
Using Apt-Cacher IP Address: Default
Disable IPv6: No
Using Interface MTU Size: Default
Using DNS Search Domain: Host
Using DNS Server Address: Host
Using MAC Address: Default
Using VLAN Tag: Default
Enable Root SSH Access: No
Enable Verbose Mode: No
Creating a Zigbee2MQTT LXC using the above default settings
 ✓ Using local for Template Storage.
 ✓ Using local for Container Storage.
 ✓ Updated LXC Template List
 ✓ LXC Container 105 was successfully created.
 ✓ Started LXC Container
 ✓ Set up Container OS
 ✓ Network Connected: 192.168.10.229 
 ✓ Internet Connected
 ✓ DNS Resolved github.com to 140.82.121.4
 ✓ Updated Container OS
 ✓ Installed Dependencies
 ✓ Set up Node.js Repository
 ✓ Installed Node.js
 - Setting up Zigbee2MQTT Repository  -
[ERROR] in line 23: exit code 0: while executing command "$@" > /dev/null 2>&1
root@CHUWI-PROXMOX:~# `

**Reply by tteck** [reply]
Run in verbose mode (Advanced settings) to help determine the underlying issue.

**Reply by mablungfr** [reply]
with verbose the script run without errors...

**Reply by tteck** [reply]
🤷🏻‍♂️

**Reply by mablungfr** [reply]
thanks ;-) i will retry in default mode tomorrow

**Reply by mablungfr** [reply]
new test this morning :

`✓ Installed Node.js
 - Setting up Zigbee2MQTT Repository  /Cloning into '/opt/zigbee2mqtt'...
|remote: Enumerating objects: 27490, done.
remote: Counting objects: 100% (6508/6508), done.
remote: Compressing objects: 100% (318/318), done.
\error: RPC failed; curl 92 HTTP/2 stream 5 was not closed cleanly: CANCEL (err 8)
error: 5615 bytes of body are still expected
fetch-pack: unexpected disconnect while reading sideband packet
fatal: early EOF
fatal: fetch-pack: invalid index-pack output
|
[ERROR] in line 40: exit code 0: while executing command $STD git clone https://github.com/Koenkk/zigbee2mqtt.git /opt/zigbee2mqtt`

**Reply by tteck** [reply]
~Try verbose mode again, to help determine the underlying issue.~ That was verbose?

**Reply by tteck** [reply]
Seems that your issue is with github

**Reply by tteck** [reply]
The error message suggests a problem with the network connection or the remote repository.

**Reply by mablungfr** [reply]
yes that was verbose

**Reply by mablungfr** [reply]
can I resume the script without restarting the complete procedure ?

**Reply by mablungfr** [reply]
With another computer with mac os, when I clone https://github.com/Koenkk/zigbee2mqtt.git I have the same error.

**Reply by tteck** [reply]
try `git clone --depth 1 https://github.com/Koenkk/zigbee2mqtt.git`

**Reply by tteck** [reply]
I setup a `dev` branch with the `--depth 1` flag, see if that works

```
bash -c "$(wget -qLO - https://github.com/tteck/Proxmox/raw/dev/ct/zigbee2mqtt.sh)"
```

**Reply by mablungfr** [reply]
ok i'm trying with default settings

**Reply by tteck** [reply, marked as answer]
It works on my end, so I'll modify the script in the main branch to use the shallow clone.

**Reply by mablungfr** [reply]
Ok no errors. Perfect, thanks for your help and your work!
